pbm/backup/oplog.go

```diff
@@ -62,13 +71,37 @@ func (ot *Oplog) WriteTo(w io.Writer) (int64, error) {
 	defer cur.Close(ctx)
 
 	opts := primitive.Timestamp{}
-	var ok bool
+	var ok, rcheck bool
 	var written int64
 	for cur.Next(ctx) {
 		opts.T, opts.I, ok = cur.Current.Lookup("ts").TimestampOK()
 		if !ok {
 			return written, errors.Errorf("get the timestamp of record %v", cur.Current)
 		}
+		// Before processing the first oplog record we check if oplog has sufficient range,
+		// i.e. if there are no gaps between the ts of the last backup or slice and
+		// the first record of the current slice. Whereas the request is >= last_saved_ts
+		// we don't know if the returned oldest record is the first since last_saved_ts or
+		// there were other records that are now removed because of oplog collection capacity.
+		// So after we retrieved the first record of the current slice we check if there is
+		// at least one preceding record (basically if there is still record(s) from the previous set).
+		// If so, we can be sure we have a contiguous history with respect to the last_saved_slice.
+		//
+		// We should do this check only after we retrieved the first record of the set. Otherwise,
+		// there is a possibility some records would be erased in a time span between check and first
+		// recorded retrieval due to ongoing write traffic (i.e. oplog append). There's a chance of
+		// false-negative though.
+		if !rcheck {
+			c, err := cl.CountDocuments(ctx, bson.M{"ts": bson.M{"$lte": ot.start}})
```

Comment:
If a count command is run it will scan all documents (or least those before the timestamp).

I suggest doing a findOne() on the same clause and that way the db server can stop scanning oplog documents as soon as possible.

Reply:
Eah, I forgot to add option `{limit: 1}` to this call. I believe it does the same - stops scanning docs after the firs find

Reply:
fixed 95b46c1

Reply:
I didn't realize there was a limit argument for count, but there we are. OK.